hazelcast/src/main/java/com/hazelcast/jet/JobInvocationObserver.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet;
+
+import com.hazelcast.internal.cluster.MemberInfo;
+import com.hazelcast.jet.config.JobConfig;
+import com.hazelcast.jet.core.DAG;
+
+import java.util.Set;
+
+public interface JobInvocationObserver {
+
+    void onJobInvocation(long jobId, Set<MemberInfo> members, DAG dag, JobConfig jobConfig);
```

Comment:
when this observer is going to be invoked (add javadoc)? Should it be invoked in case of failures during job startup (currently it it invoked late)?

Reply:
it is also not invoked for normal jobs but you did not documented that fact. This is highly unexpected behavior and trap for us.

Reply:
Good catch, thanks.

Reply:
Done.